dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,287 @@
+            .withArguments("assemble")
+            .withPluginClasspath()
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M success W assemble { build cache }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("--build-cache", "assemble")
+            .withPluginClasspath()
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M fail W assemble { configuration cache, checkProjectDependecies enabled }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        assertThatThrownBy {
+            GradleRunner.create()
+                .withProjectDir(testProjectDir)
+                .withArguments("--configuration-cache", "assemble")
+                .withPluginClasspath()
+                .build()
+        }.isInstanceOf(Exception::class.java)
+    }
+
+    // The assembleRelease fails for configuration cache and not because of our code but because of
+    // these tasks: collect[Flavour1][Flavour2]ReleaseDependencies
+    @Test
+    fun `M success W assembleDebug { configuration cache, checkProjectDependencies disabled }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_check_deps_disabled.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("--configuration-cache", "assembleDebug")
+            .withPluginClasspath()
+            .build()
+
+        // Then
+        assertThat(result.task(":assembleDebug")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M success W assemble { Datadog SDK not in deps, checkProjectDepedencies set to warn }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_check_deps_set_to_warn.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("assemble")
+            .withPluginClasspath()
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M fail W assemble { Datadog SDK not in deps }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_without_datadog_dep.gradle.kts")
+        // When
+        assertThatThrownBy {
+            GradleRunner.create()
+                .withProjectDir(testProjectDir)
+                .withArguments("assemble")
+                .withPluginClasspath()
+                .build()
+        }.isInstanceOf(Exception::class.java)
```

Comment:
Again can we make a more precise check on which exception we're expecting here?

Reply:
Not so sure as it is a generic exception but will try to have a look

Reply:
Unfortuantely I cannot get the root cause of this exception as `GradleRunner` is intercepting that but it is not passing it along :(